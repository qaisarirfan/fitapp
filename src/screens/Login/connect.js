import {connect} from "react-redux"
import {login} from "../../redux/reducers/authentication/actionCreators"

const mapStateToProps = (state) => ({})

const mapDispatchToProps = {
  login,
}

export default connect(mapStateToProps, mapDispatchToProps)
