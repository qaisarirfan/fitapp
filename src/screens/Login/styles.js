import Color from "color"
import {Dimensions, StyleSheet} from "react-native"

const screen = Dimensions.get("window")

// Login: Styling here
const styles = () => {
  return StyleSheet.create({
    root: {
      flex: 1,
      backgroundColor: "#2ecc71",
    },
    container: {},
    content: {
      paddingTop: screen.height * 0.04,
    },
    welcomeBack: {
      fontSize: 44,
      color: "rgb(255, 255, 255)",
      fontWeight: "bold",
      textAlign: "center",
      fontFamily: "Roboto",
    },

    inputContainer: {},
    input: {
      padding: 14,
      paddingLeft: 10,
      paddingRight: 10,
      fontSize: 24,
      // color: Color(theme.A700).darken(0.6).hex(),
      backgroundColor: "transparent",
      // marginVertical: 10,
      flexDirection: "row",
      borderBottomWidth: StyleSheet.hairlineWidth,
      borderBottomColor: "#ffffff",
      textAlign: "center",
      color: "rgb(255, 255, 255)",
    },
    submitButton: {
      backgroundColor: "#fff",
      margin: 30,
      alignItems: "center",
      borderRadius: 30,
      overflow: "hidden",
    },
    submitButtonText: {
      color: "#2ecc71",
      fontSize: 18,
      padding: 10,
      paddingBottom: 16,
      paddingTop: 16,
    },
  })
}
export default styles
