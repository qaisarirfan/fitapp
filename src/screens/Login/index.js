import React, {useState} from "react"
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  Alert,
  ActivityIndicator,
} from "react-native"
import flow from "lodash/flow"
import isEqual from "lodash/isEqual"
import PropTypes from "prop-types"
import {useRoute} from "@react-navigation/native"

import {KeyboardAwareScrollView} from "react-native-keyboard-aware-scroll-view"
import {Formik} from "formik"
import * as Yup from "yup"
import {HeaderBar} from "../../components/HeaderBar"

import {Logo} from "../../components/Logo"
import Tooltip from "../../components/Tooltip"
import themeStyles from "./styles"
import connect from "./connect"

// Login Component content
export const Login = ({login}) => {
  const styles = themeStyles()
  const {name} = useRoute()

  const [loader, setLoader] = useState(false)

  const sleep = (duration) =>
    new Promise((resolve) => setTimeout(() => resolve(), duration))

  return (
    <View style={styles.root} testID="login_screen">
      <HeaderBar isHeaderShow={false} title={name} />
      <KeyboardAwareScrollView
        scrollEnabled
        behavior="padding"
        style={styles.container}
        contentContainerStyle={styles.container}>
        <View style={styles.content}>
          <Logo />
          <Text style={styles.welcomeBack}>Welcome back!</Text>
          <Formik
            initialValues={{email: "", password: ""}}
            validationSchema={Yup.object().shape({
              email: Yup.string().email("Invalid email").required("Required"),
              password: Yup.string().required("Required"),
            })}
            onSubmit={(values) => {
              const loginFakeData = {
                email: "admin@admin.com",
                password: "admin",
              }
              if (isEqual(values, loginFakeData)) {
                setLoader(true)
                sleep(5000).then(() => {
                  login()
                  setLoader(false)
                })
              } else {
                Alert.alert(
                  "Error",
                  "The email address that you've entered doesn't match any account.",
                  [
                    {
                      text: "OK",
                      onPress: () => console.log("OK Pressed"),
                      testID: "alert",
                    },
                  ]
                )
              }
            }}>
            {({
              handleChange,
              handleBlur,
              handleSubmit,
              values,
              errors,
              touched,
            }) => (
              <View style={styles.inputContainer}>
                <View>
                  <TextInput
                    style={styles.input}
                    placeholder="Email"
                    placeholderTextColor={"#fff"}
                    autoCapitalize="none"
                    autoCompleteType="email"
                    value={values.email}
                    onChangeText={handleChange("email")}
                    onBlur={handleBlur("email")}
                    testID="email"
                  />
                  {touched.email && errors.email ? (
                    <Text>{errors.email}</Text>
                  ) : null}
                </View>
                <View>
                  <TextInput
                    style={styles.input}
                    placeholder="Password"
                    placeholderTextColor={"#fff"}
                    autoCapitalize="none"
                    secureTextEntry
                    value={values.password}
                    onChangeText={handleChange("password")}
                    onBlur={handleBlur("password")}
                    testID="password"
                  />
                  {touched.password && errors.password ? (
                    <Text>{errors.password}</Text>
                  ) : null}
                </View>
                <TouchableOpacity
                  style={styles.submitButton}
                  onPress={handleSubmit}
                  testID="submit">
                  <Text style={styles.submitButtonText}>Submit</Text>
                </TouchableOpacity>
                {loader ? <ActivityIndicator color="#fff" /> : null}
              </View>
            )}
          </Formik>
        </View>
      </KeyboardAwareScrollView>
    </View>
  )
}

// Login Proptypes
Login.propTypes = {
  login: PropTypes.func.isRequired,
}

// Login Default props
Login.defaultProps = {}

export default flow([connect])(Login)
