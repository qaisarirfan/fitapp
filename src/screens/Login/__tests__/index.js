import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import Login from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<Login />", () => {
  const createProps = () => ({})
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<Login {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
