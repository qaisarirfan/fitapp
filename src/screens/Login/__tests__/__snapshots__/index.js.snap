// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4f6d7a",
      "flex": 1,
    }
  }
  testID="login_screen"
>
  <View
    style={
      Array [
        Object {},
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
            "paddingLeft": 6,
            "width": 250,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "center",
            "paddingRight": 6,
            "width": 250,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    behavior="padding"
    contentContainerStyle={Object {}}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={Object {}}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "paddingTop": 53.36,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "marginBottom": 40,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 300,
                "width": 300,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginHorizontal": 20,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCompleteType="email"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Email"
            placeholderTextColor="#00476E"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderColor": "#a7b6bd",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "color": "#00476E",
                "flexDirection": "row",
                "fontSize": 18,
                "marginVertical": 10,
                "padding": 14,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
            testID="email"
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Password"
            placeholderTextColor="#00476E"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderColor": "#a7b6bd",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "color": "#00476E",
                "flexDirection": "row",
                "fontSize": 18,
                "marginVertical": 10,
                "padding": 14,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
            testID="password"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#263e4a",
                "borderRadius": 20,
                "height": 40,
                "margin": 15,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 10,
              }
            }
            testID="submit"
          >
            <Text
              style={
                Object {
                  "color": "#eaedef",
                  "fontSize": 18,
                }
              }
            >
              Submit
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "right": 0,
              },
              Object {
                "height": 0,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
